src/wasm-lib/kcl/src/walk/ast_node.rs

```diff
@@ -76,6 +76,48 @@ impl Node<'_> {
             Node::LabelledExpression(n) => n.digest,
         }
     }
+
+    /// Check to see if this [Node] points to the same underlying specific
+    /// borrowed object as another [Node]. This is not the same as `Eq` or
+    /// even `PartialEq` -- anything that is `true` here is absolutely `Eq`,
+    /// but it's possible this node is `Eq` to another with this being `false`.
+    ///
+    /// This merely indicates that this [Node] specifically is the exact same
+    /// borrowed object as [Node].
+    pub fn ptr_eq(&self, other: Node) -> bool {
+        unsafe { std::ptr::eq(self.ptr(), other.ptr()) }
+    }
+
+    unsafe fn ptr(&self) -> *const () {
+        match self {
+            Node::Program(n) => *n as *const _ as *const (),
```

Comment:
yeah i know, 🤮

Reply:
If you really wanted to be fancy you could make a macro to reduce duplication of the nasty, but it's prob not a big deal.